docs/Code_Quality_Guidelines.rst

```diff
@@ -0,0 +1,68 @@
+
+    if packets_mode not in ["virtual", "real"]:
+            raise ValueError(
+                "Invalid value passed to packets_mode. Only "
+                "allowed values are 'virtual' or 'real'"
+            )
+In this example of code, any invalid input triggers the conditional statement, and it raises a specific error (ValueError) and gives a message that informs the user what their error is and the appropriate values that can be used to avoid this error. In this manner, most edge case inputs and possible errors can be caught so the user can address them. This practice allows for specific errors to be avoided to make debugging easier and makes it so edge cases of invalid input are caught.
+
+Docstrings
+==========
+
+For docstrings, TARDIS follows the docstring formatting of `Numpydocs <https://numpydoc.readthedocs.io/en/latest/format.html>`_. 
+This is an example of a properly formatted docstring::
+
+    """
+    Read different density file formats
+
+    Parameters
+    ----------
+    `filename` : str
+        filename or path of the density file
+    `filetype` : str
+        type of the density file
+
+    Returns
+    -------
+    `time_of_model` : astropy.units.Quantity
+        time at which the model is valid
+    `velocity` : np.ndarray
+        the array containing the velocities
+    `unscaled_mean_densities` : np.ndarray
+        the array containing the densities
+    """
+
+The docstring should have no leading or trailing carriage returns, and there should be a carriage return between each segment. At the start of the docstring there is a summary explaining the purpose of the function/class/module. This summary should follow the standard English syntax, starting with a capitalized letter and ending with appropriate punctuation. This is necessary for every docstring. The docstring summary should not explain individual lines or the returns, it should summarize the purpose of the function/class/module. Comments on how individual lines work should be done using in-line comments. 
+
+From numpy's Common reST concepts: "Variable, module, function, and class names should be written between single back-ticks ` `". Please use this syntax in docstrings. In the above example the return variable and type is specified. For the Returns, the type must always be stated, even if the variable is not. The Returns section should follow the format of::
+
+    Returns
+    -------
+    (`optional variable name` : )type
+        (optional descriptor)
+
+In the first example, the full path can be shown for the return time_of_model. Always list the full path for a variable type if it is not a built-in type. The Returns section should not be included if the function/module/class does not have a return value(s).
+
+Naming Conventions
```

Comment:
This naming convention is actually part of Code Style Guide - so should be merged with it